From bf41fa026ae3d378e62fd83d03a6f5933b52ca04 Mon Sep 17 00:00:00 2001
From: Alistair Francis <alistair@alistair23.me>
Date: Thu, 14 Nov 2019 13:08:31 -0800
Subject: [PATCH] meson.build: make TLS ELF optional

USE_ELF_TLS has replaced GLX_USE_TLS so this patch is the original "make
TLS GLX optional again" patch updated to the latest mesa.

For details, see:
https://gitlab.freedesktop.org/mesa/mesa/-/issues/966

This prevents runtime segfault on musl:

Traceback (most recent call last):
  File "/home/pokybuild/yocto-worker/musl-qemux86/build/meta/lib/oeqa/core/decorator/__init__.py", line 36, in wrapped_f
    return func(*args, **kwargs)
  File "/home/pokybuild/yocto-worker/musl-qemux86/build/meta/lib/oeqa/runtime/cases/parselogs.py", line 378, in test_parselogs
    self.assertEqual(errcount, 0, msg=self.msg)
AssertionError: 1 != 0 : Log: /home/pokybuild/yocto-worker/musl-qemux86/build/build/tmp/work/qemux86-poky-linux-musl/core-image-sato-sdk/1.0-r0/target_logs/Xorg.0.log

Upstream-Status: Inappropriate [configuration]
---
 meson.build       | 7 +++++--
 meson_options.txt | 6 ++++++
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 70d06c0..1441611 100644
--- a/meson.build
+++ b/meson.build
@@ -490,8 +490,11 @@ foreach platform : _platforms
   pre_args += '-DHAVE_@0@_PLATFORM'.format(platform.to_upper())
 endforeach
 
-use_elf_tls = true
-pre_args += '-DUSE_ELF_TLS'
+use_elf_tls = false
+if get_option('elf-tls')
+  use_elf_tls = true
+  pre_args += '-DUSE_ELF_TLS'
+endif
 
 if with_platform_android and get_option('platform-sdk-version') >= 29
   # By default the NDK compiler, at least, emits emutls references instead of
diff --git a/meson_options.txt b/meson_options.txt
index 1f6ef38..99cc5cb 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -440,6 +440,12 @@ option(
   value : true,
   description : 'Enable direct rendering in GLX and EGL for DRI',
 )
+option(
+  'elf-tls',
+  type : 'boolean',
+  value : true,
+  description : 'Enable TLS support in ELF',
+)
 option('egl-lib-suffix',
   type : 'string',
   value : '',
